apps/projects/views.py

```diff
@@ -294,49 +298,87 @@ def raise_exception(self):
 
     def get_context_data(self, **kwargs):
         """Append insights to the template context."""
+
         context = super().get_context_data(**kwargs)
+        idea_blueprints = {"BS", "IC"}
+        map_idea_blueprints = {"MBS", "MIC", "PB"}
 
         modules = self.project.modules.all()
+        active_blueprints = set(modules.values_list("blueprint_type", flat=True))
+
+        enable_ideas = active_blueprints.intersection(idea_blueprints)
+        enable_map_ideas = active_blueprints.intersection(map_idea_blueprints)
+        enable_polls = "PO" in active_blueprints
+        enable_interactive_event = "IE" in active_blueprints
+
         ideas = Idea.objects.filter(module__in=modules)
         map_ideas = MapIdea.objects.filter(module__in=modules)
-        proposals = Proposal.objects.filter(module__in=modules)
         comments = get_all_comments_project(project=self.project)
 
-        ratings_ideas = Rating.objects.filter(idea__in=ideas)
-        ratings_map_ideas = Rating.objects.filter(mapidea__in=map_ideas)
-        ratings_comments = Rating.objects.filter(comment__in=comments)
+        polls = Poll.objects.filter(module__in=modules)
+        votes = Vote.objects.filter(choice__question__poll__in=polls)
+        answers = Answer.objects.filter(question__poll__in=polls)
+        live_questions = LiveQuestion.objects.filter(module__in=modules)
+        likes = Like.objects.filter(livequestion__in=live_questions)
+
+        values = [Rating.POSITIVE, Rating.NEGATIVE]
+        ratings_ideas = Rating.objects.filter(idea__in=ideas, value__in=values)
+        ratings_map_ideas = Rating.objects.filter(
+            mapidea__in=map_ideas, value__in=values
+        )
+        ratings_comments = Rating.objects.filter(comment__in=comments, value__in=values)
+
```

Comment:
The ratings on topics are missing (topic prioritization module), probably also a thing to add below with the flags?

Reply:
yes! will do